src/midifunctor.cpp

```diff
@@ -313,6 +317,170 @@ FunctorCode InitTimemapTiesFunctor::VisitTie(Tie *tie)
+                double startRealTimeSeconds = startTime.ToDouble() * 60.0 / m_currentTempo;
+                double stopRealTimeSeconds = stopTime.ToDouble() * 60.0 / m_currentTempo;
+                note->SetRealTimeOnsetSeconds(startRealTimeSeconds);
+                note->SetRealTimeOffsetSeconds(stopRealTimeSeconds);
+            }
+            startTime = stopTime;
+        }
+
+        m_graces.clear();
+    }
+
+    return FUNCTOR_CONTINUE;
+}
+
+FunctorCode InitTimemapNotesFunctor::VisitMeasure(Measure *measure)
+{
+    // Update the current tempo
+    m_currentTempo = measure->GetCurrentTempo();
+
+    return FUNCTOR_CONTINUE;
+}
+
+FunctorCode InitTimemapNotesFunctor::VisitNote(Note *note)
+{
+    // Skip linked notes
+    if (note->HasSameasLink()) {
+        return FUNCTOR_SIBLINGS;
+    }
+
+    // Skip cue notes when midiNoCue is activated
+    if ((note->GetCue() == BOOLEAN_true) && m_noCue) {
+        return FUNCTOR_SIBLINGS;
+    }
+
+    // If the note is a secondary tied note, then ignore it
+    if (note->GetScoreTimeTiedDuration() < 0) {
+        return FUNCTOR_SIBLINGS;
+    }
+
+    // Handle grace notes
+    if (note->IsGraceNote()) {
+        m_accentedGraceNote = (note->GetGrace() == GRACE_acc);
+        double time = note->HasGraceTime() ? note->GetGraceTime() : 50.0;
+        const GraceGrp *graceGrp = vrv_cast<const GraceGrp *>(note->GetFirstAncestor(GRACEGRP));
+        if (graceGrp) {
+            if (graceGrp->GetGrace() == GRACE_acc) m_accentedGraceNote = true;
+            time = (graceGrp->HasGraceTime()) ? graceGrp->GetGraceTime() : 50.0;
+        }
+        m_graces.push_back({ { note }, note->GetDur(), time });
+
+        return FUNCTOR_SIBLINGS;
+    }
+
+    // Check if some grace notes must be performed
+    if (!m_graces.empty()) {
+        this->AddGraceNotesFor(note);
+        m_graces.clear();
+    }
+
+    // Store reference, i.e. for Nachschlag
+    m_lastNote = note;
+
+    return FUNCTOR_SIBLINGS;
+}
+
+void InitTimemapNotesFunctor::AddGraceNotesFor(Note *refNote)
+{
+    Fraction startTime = refNote->GetScoreTimeOnset();
+
+    Fraction graceNoteDur = 0;
+    if (m_accentedGraceNote && !m_graces.empty()) {
+        // Arbitraritly looks at the first note, not sure what to do if we have condradictory values
+        double percent = m_graces.front().time;
+        // Arbitraritly constraint the time between 5% and 95%
+        percent = std::min(95.0, std::max(5.0, percent));
+        const Fraction totalDur = refNote->GetScoreTimeDuration() * (int)percent / 100;
+        // Adjust the start of the main note
+        refNote->SetScoreTimeOnset(startTime + totalDur);
+        double startRealTime = (startTime + totalDur).ToDouble() * 60.0 / m_currentTempo;
+        refNote->SetRealTimeOnsetSeconds(startRealTime);
+        graceNoteDur = totalDur / (int)m_graces.size();
+    }
+    else {
+        graceNoteDur = UNACC_GRANENOTE_FRACTION * (int)m_currentTempo / 120;
```

Comment:
Same issue as above (L362): `graceNoteDur` is three times shorter than before, such that unaccented grace notes are played perceivably faster.